### PR TITLE
tests(smokehouse): fail on finalUrl/errorCode mismatches

### LINKED ISSUE #7225: Smokehouse tests do not fail on finalUrl/errorCode mismatches
```
diff --git a/lighthouse-cli/test/smokehouse/pwa-expectations.js b/lighthouse-cli/test/smokehouse/pwa-expectations.js
index 9db56360..3ebd2c37 100644
--- a/lighthouse-cli/test/smokehouse/pwa-expectations.js
+++ b/lighthouse-cli/test/smokehouse/pwa-expectations.js
@@ -24,7 +24,7 @@ const pwaDetailsExpectations = {
 module.exports = [
   {
     requestedUrl: 'https://airhorner.com',
-    finalUrl: 'https://airhorner.com/',
+    finalUrl: 'https://airhorner.com/asdasdasdas',
     audits: {
       'is-on-https': {
         score: 1,

```

image

Note that the status code is 0.

This is due to the way errors are counted in `smokehouse.js report()`.

### PR DESCRIPTION
**Pull request by Hoten**
Fixes #7225.

### CODE REVIEW

File: lighthouse-cli/test/smokehouse/smokehouse.js [hunk 1]
```diff
@@ -311,11 +311,23 @@ function reportAssertion(assertion) {
...
   reportAssertion(results.errorCode);
+  if (results.errorCode.equal) {
+    correctCount++;
+  } else {
+    failedCount++;
+  }
 
-  let correctCount = 0;
-  let failedCount = 0;
   results.audits.forEach(auditAssertion => {
```
Comment (paulirish) [inline lighthouse-cli/test/smokehouse/smokehouse.js]:
```suggestion
  [results.finalUrl, results.errorCode, ...results.audits].forEach(auditAssertion => {
```

Comment (paulirish) [reply lighthouse-cli/test/smokehouse/smokehouse.js]: could do this instead and drop the two reportAssertions above.

Comment (Hoten) [reply lighthouse-cli/test/smokehouse/smokehouse.js]: nice

do we want to continue showing green ticks for passing? this is what it looks like  with those changes + moving `reportAssertion` out of the if statement:

image


Keeping `reportAssertion` where it is-

image

Comment (Hoten) [reply lighthouse-cli/test/smokehouse/smokehouse.js]: i like this best

image

**Comment by patrickhulce** [comment]
I kinda like the verbose version. My most common use case when actually running smoketests is picking a particular smoke test and running it frequently so I kinda like the ability to see what's passing. I have no strong feelings on the length of passing CI logs since those are collapsed for me anyhow :)

**Comment by Hoten** [comment]
`--verbose` it is!

**Comment by Hoten** [comment]
Went with an env flag since wiring an option would be more work (see how run-smokehouse parses argv).

File: lighthouse-cli/test/smokehouse/smokehouse.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ const PAGE_HUNG_EXIT_CODE = 68;
 const INSECURE_DOCUMENT_REQUEST_EXIT_CODE = 69;
 const RETRIES = 3;
 const NUMERICAL_EXPECTATION_REGEXP = /^(<=?|>=?)((\d|\.)+)$/;
+const VERBOSE = process.env.LH_SMOKE_VERBOSE === '1';
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/smokehouse.js]: WDYT about just `Boolean(process.env.LH_SMOKE_VERBOSE)`?

Comment (patrickhulce) [reply lighthouse-cli/test/smokehouse/smokehouse.js]: later down we use a env var `SMOKEHOUSE_DEBUG` maybe we should unify these two 🤔 

maybe we rename that one to `LH_SMOKE_DEBUG`?

File: lighthouse-cli/test/smokehouse/smokehouse.js [hunk 3]
```diff
@@ -276,8 +277,12 @@ function reportAssertion(assertion) {
   RegExp.prototype.toJSON = RegExp.prototype.toString;
 
   if (assertion.equal) {
-    console.log(`  ${log.greenify(log.tick)} ${assertion.category}: ` +
-        log.greenify(assertion.actual));
+    if (assertion.actual !== null && typeof assertion.actual === 'object') {
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/smokehouse.js]: super nit but mind flipping the order of these to `is an object && is not null`? took me a bit to realize this was just an object check

Comment (paulirish) [reply lighthouse-cli/test/smokehouse/smokehouse.js]: can we just use a helper method like so?  

```js
const isPlainObject = function (obj) {
	return Object.prototype.toString.call(obj) === '[object Object]';
};
```

File: lighthouse-cli/test/smokehouse/smokehouse.js [hunk 4]
```diff
@@ -62,7 +63,7 @@ function resolveLocalOrCwd(payloadPath) {
  * @return {ExpectedLHR}
  */
 function runLighthouse(url, configPath, isDebug) {
-  isDebug = isDebug || Boolean(process.env.SMOKEHOUSE_DEBUG);
+  isDebug = isDebug || Boolean(process.env.LH_SMOKE_DEBUG);
```
Comment (brendankenny) [inline lighthouse-cli/test/smokehouse/smokehouse.js]: doesn't debug also save the LHR? Up to y'all if you want that :)

File: lighthouse-cli/test/smokehouse/smokehouse.js (hunk 3 above)
Comment (brendankenny) [reply lighthouse-cli/test/smokehouse/smokehouse.js]: > `Object.prototype.toString.call(obj) === '[object Object]';`

what year is it?

Comment (brendankenny) [reply lighthouse-cli/test/smokehouse/smokehouse.js]: :P

File: lighthouse-cli/test/smokehouse/smokehouse.js (hunk 4 above)
Comment (Hoten) [reply lighthouse-cli/test/smokehouse/smokehouse.js]: nah. just writing to disk is good enough for me. verbose shouldn't be _too_ verbose

**Pull request merged**